### PR TITLE
Add -h as alias to --help

### PR DESCRIPTION
**Pull request by hustLer2k**
### What does this PR do?

Implements https://github.com/oven-sh/bun/issues/7451.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

_**Existing tests pass and manual testing.**_

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I included a test for the new code, or an existing test covers it
- [x] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [x] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: src/cli.zig [hunk 1]
```diff
@@ -368,7 +368,7 @@ pub const Arguments = struct {
             Global.exit(1);
         };
 
-        const print_help = args.flag("--help");
+        const print_help = args.flag("--help") or args.flag("-h");
```
Comment (paperclover) [inline src/cli.zig]: did you check if the `or args.flag("-h")` is needed. i think clap is able to do this implicitly once you speicfy the -h alias  in the param list

Comment (hustLer2k) [reply src/cli.zig]: it's actually not needed

File: src/cli.zig [hunk 2]
```diff
@@ -1515,7 +1515,7 @@ pub const Command = struct {
                 const print_help = brk: {
                     for (bun.argv()) |arg_| {
                         const arg = bun.span(arg_);
-                        if (strings.eqlComptime(arg, "--help")) {
+                        if (strings.eqlComptime(arg, "--help") or strings.eqlComptime(arg, "-h")) {
```
Comment (paperclover) [inline src/cli.zig]: unneeded

**Review by paperclover — CHANGES_REQUESTED** [review]

File: src/cli/init_command.zig [hunk 3]
```diff
@@ -90,7 +90,7 @@ pub const InitCommand = struct {
         const print_help = brk: {
             for (argv) |arg_| {
                 const arg = bun.span(arg_);
-                if (strings.eqlComptime(arg, "--help")) {
+                if (strings.eqlComptime(arg, "--help") or strings.eqlComptime(arg, "-h")) {
```
Comment (paperclover) [inline src/cli/init_command.zig]: unneeded x2

**Comment by paperclover** [comment]
looks great besides hte unneeded ones

**Review by paperclover — APPROVED** [review]
looks great. will merge after tests

**Pull request merged**